### PR TITLE
Input component

### PR DESCRIPTION
**Pull request by kterekh**
Task: https://trello.com/c/xe7YAOQP/13-create-input-component

Please review.

### CODE REVIEW

File: src/common/Button.vue [hunk 1]
```diff
@@ -16,6 +16,7 @@ export default class Button extends Vue {
   @Prop({ required: true }) title!: string
   @Prop({ default: false }) isDisabled!: boolean
   @Prop() action!: string
+  @Prop({ default: 'button' }) type!: string
```
Comment (silchencko) [inline src/common/Button.vue:19]: Could you create an enum for button types, please?

File: src/common/Button.vue [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <button type="button" class="btn" v-bind:class="btnClass" v-bind:disabled="isDisabled" v-on:click="action">{{
+  <button :type="type" class="btn" v-bind:class="btnClass" v-bind:disabled="isDisabled" v-on:click="action">{{
```
Comment (silchencko) [inline src/common/Button.vue:2]:
```suggestion
  <button :type="type" 
          class="btn" 
          :class="btnClass" 
          :disabled="isDisabled" 
          @click="action">{{
```

File: src/common/Input.vue [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+</template>
+
+<script lang="ts">
+import { Component, Vue, Prop } from 'vue-property-decorator'
+import { Validation } from 'vuelidate'
+
+@Component
+export default class Input extends Vue {
+  value = ''
+  @Prop() type!: string
```
Comment (silchencko) [inline src/common/Input.vue:25]: Could you create an enum for Input type, please?

File: src/common/TextArea.vue [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+</template>
+
+<script lang="ts">
+import { Component, Vue, Prop } from 'vue-property-decorator'
+import { Validation } from 'vuelidate'
+
+@Component
+export default class Input extends Vue {
+  value = ''
+  @Prop() type!: string
```
Comment (silchencko) [inline src/common/TextArea.vue:28]: Do you need `type` here?

File: src/common/TextArea.vue [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  set email (value) {
+    this.v.$touch()
+    this.$emit('input', value)
+  }
+}
+</script>
+
+<style lang="scss">
+.form-group--error {
+  border: 2px solid red !important;
```
Comment (silchencko) [inline src/common/TextArea.vue:49]: My suggestion is to move `2px solid red !important;` to scss variable or `border: 2px solid red !important;` to scss mixin

File: src/components/FormDemoComponent.vue [hunk 6]
```diff
@@ -0,0 +1,67 @@
+<template>
+  <form @submit.prevent="submit">
+    <Input
+      :v="$v.form.email"
+      v-model="form.email"
+      v-bind:is-disabled="false"
```
Comment (silchencko) [inline src/components/FormDemoComponent.vue:6]: I recommend to use one style: `v-bind:` or `:`

File: src/common/TextArea.vue [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Prop } from 'vue-property-decorator'
+import { Validation } from 'vuelidate'
+
+@Component
+export default class Input extends Vue {
+  value = ''
```
Comment (silchencko) [inline src/common/TextArea.vue:27]:
```suggestion
  value: string = ''
```

File: src/common/TextArea.vue [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  @Prop() type!: string
+  @Prop({ default: 'false' }) isDisabled!: boolean
+  @Prop() v!: Validation
+
+  get commentError (): boolean {
+    return this.v.$error
+  }
+
+  get email () {
```
Comment (silchencko) [inline src/common/TextArea.vue:36]:
```suggestion
  get email (): string {
```

File: src/common/Input.vue [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Prop } from 'vue-property-decorator'
+import { Validation } from 'vuelidate'
+
+@Component
+export default class Input extends Vue {
+  value = ''
```
Comment (silchencko) [inline src/common/Input.vue:24]:
```suggestion
  value: string = ''
```

File: src/common/Input.vue [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+  @Prop() type!: string
+  @Prop({ default: 'false' }) isDisabled!: boolean
+  @Prop() v!: Validation
+
+  get emailError (): boolean {
+    return this.v.$error
+  }
+
+  get email () {
```
Comment (silchencko) [inline src/common/Input.vue:33]:
```suggestion
  get email (): string {
```

**Review by silchencko — COMMENTED** [review]
Big work done! My congratulations! 
My suggestion is to add more different input types on you form. It will help you to make your `Input` component more flexible. Now it supports only email and required errors. But input text can have min or max length, repeat password should be the same as password etc.

File: src/common/Button.vue (hunk 1 above)
Comment (kterekh) [reply src/common/Button.vue:19]: Make sense.

File: src/common/Button.vue (hunk 2 above)
Comment (kterekh) [reply src/common/Button.vue:2]: Follows the previous comment. Thanks.

File: src/common/Input.vue (hunk 3 above)
Comment (kterekh) [reply src/common/Input.vue:25]: Yes, sure.

File: src/common/TextArea.vue (hunk 4 above)
Comment (kterekh) [reply src/common/TextArea.vue:28]: No, was copied from input component accidentally.

File: src/components/FormDemoComponent.vue (hunk 6 above)
Comment (kterekh) [reply src/components/FormDemoComponent.vue:6]: Thanks! Changed through all files.

File: src/common/TextArea.vue (hunk 7 above)
Comment (kterekh) [reply src/common/TextArea.vue:27]: Screenshot_3

File: src/common/Input.vue [hunk 11]
```diff
@@ -1,36 +1,46 @@
...
+  get inputError (): boolean {
     return this.v.$error
   }
 
-  get email () {
+  inputValidation (param: string): boolean {
+    return !this.v[param] && this.v.$params[param]
+  }
+
+  get email (): string {
```
Comment (silchencko) [inline src/common/Input.vue:43]:
```suggestion
  get input_value (): string {
```

Comment (silchencko) [reply src/common/Input.vue:43]: set input_value

**Review by silchencko — APPROVED** [review]

**Pull request merged**